pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,49 @@
+package declarativeconfig
+
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+type ClaimMapping struct {
+	PathToClaim    string `yaml:"pathToClaim,omitempty"`
+	MapToClaimName string `yaml:"mapTo,omitempty"`
+}
+
+type OIDCConfig struct {
+	Issuer                    string `yaml:"issuer,omitempty"`
+	CallbackMode              string `yaml:"mode,omitempty"`
+	ClientID                  string `yaml:"clientID,omitempty"`
+	ClientSecret              string `yaml:"clientSecret,omitempty"`
+	DisableOfflineAccessScope bool   `yaml:"disableOfflineAccessScope,omitempty"`
+	DontUseClientSecretConfig bool   `yaml:"dontUseClientSecretConfig,omitempty"`
+}
+
+type SAMLConfig struct {
+	// TODO: add more fields
+	// "sp_issuer"
+	// "idp_metadata_url"
+	// "idp_cert_pem"
+	// "idp_sso_url"
+	// "idp_nameid_format"
+	SPIssuer string `yaml:"spIssuer,omitempty"`
+}
+
+// AuthProvider is representation of storage.PermissionSet that supports transformation from YAML.
+type AuthProvider struct {
+	Name               string              `yaml:"name,omitempty"`
+	MinimumRoleName    string              `yaml:"minimumRole,omitempty"`
```

Comment:
> Either we handle this within the struct (if miminumRoleName == "" -> do not create default group or within the configuration). 

By handling within the struct you probably mean within the  `declarative config Go struct -> proto struct` transformation logic?

 Right now, whatever we do, I think we should just keep this field as a string and not change YAML -> Go struct transformation logic. I would keep YAML -> declarative config Go struct transformation as simple as possible.

>  I think we should handle it within the struct itself, leave a comment specifying that, if left empty, no default group will be created.

Let's leave a TODO if we have ticket for transformation/validation of declarative config structs.

> (On an unrelated note, it might be good to add some validation to some of the structs in a basic manner, either directly or via things such as [go-playground/validator](https://github.com/go-playground/validator) which we already seem to have as an indirect dependency)

100%

Reply:
Ticket for transformation is [ROX-14148](https://issues.redhat.com/browse/ROX-14148)